packages/astro/test/react-component.test.js

```diff
@@ -4,98 +4,125 @@ import { loadFixture } from './test-utils.js';
-    expect($('#react-h2').text()).to.equal('Hello world!');
+    it('Can load React', async () => {
+      const html = await fixture.readFile('/index.html');
+      const $ = cheerio.load(html);
 
-    // test 2: no reactroot
-    expect($('#react-h2').attr('data-reactroot')).to.equal(undefined);
+      // test 1: basic component renders
+      expect($('#react-h2').text()).to.equal('Hello world!');
 
-    // test 3: Can use function components
-    expect($('#arrow-fn-component')).to.have.lengthOf(1);
+      // test 2: no reactroot
+      expect($('#react-h2').attr('data-reactroot')).to.equal(undefined);
 
-    // test 4: Can use spread for components
-    expect($('#component-spread-props')).to.have.lengthOf(1);
+      // test 3: Can use function components
+      expect($('#arrow-fn-component')).to.have.lengthOf(1);
 
-    // test 5: spread props renders
-    expect($('#component-spread-props').text(), 'Hello world!');
+      // test 4: Can use spread for components
+      expect($('#component-spread-props')).to.have.lengthOf(1);
 
-    // test 6: Can use TS components
-    expect($('.ts-component')).to.have.lengthOf(1);
+      // test 5: spread props renders
+      expect($('#component-spread-props').text(), 'Hello world!');
 
-    // test 7: Can use Pure components
-    expect($('#pure')).to.have.lengthOf(1);
-  });
+      // test 6: Can use TS components
+      expect($('.ts-component')).to.have.lengthOf(1);
 
-  // TODO: fix compiler bug
-  it.skip('Includes reactroot on hydrating components', async () => {
-    const html = await fixture.readFile('/index.html');
-    const $ = cheerio.load(html);
+      // test 7: Can use Pure components
+      expect($('#pure')).to.have.lengthOf(1);
+    });
 
-    const div = $('#research');
+    it('Can load Vue', async () => {
+      const html = await fixture.readFile('/index.html');
+      const $ = cheerio.load(html);
+      expect($('#vue-h2').text()).to.equal('Hasta la vista, baby');
+    });
 
-    // test 1: has the hydration attr
-    expect(div.attr('data-reactroot')).to.be.ok;
+    it('Can use a pragma comment', async () => {
+      const html = await fixture.readFile('/pragma-comment/index.html');
+      const $ = cheerio.load(html);
 
-    // test 2: renders correctly
-    expect(div.html()).to.equal('foo bar <!-- -->1');
-  });
+      // test 1: rendered the PragmaComment component
+      expect($('.pragma-comment')).to.have.lengthOf(2);
+    });
 
-  // TODO: Vite does not throw a helpful error message on window SSR
-  it.skip('Throws helpful error message on window SSR', async () => {
-    const html = await fixture.readFile('/window/index.html');
-    expect(html).to.include(
-      `[/window]
-    The window object is not available during server-side rendering (SSR).
-    Try using \`import.meta.env.SSR\` to write SSR-friendly code.
-    https://docs.astro.build/reference/api-reference/#importmeta`
-    );
-  });
+    // TODO: is this still a relevant test?
```

Comment:
Hmm, it seems like this should still be a useful test?

Test #2 checks that `data-reactroot` isn't added to the test page's `<Hello />` component since it isn't using client hydration.  It looks like this test would be making sure that `data-reactroot` is still used on the hydrated `<Research />` component

What I'm not actually sure on is whether `data-reactroot` is still used to flag hydrated react components.  If yes, I think this and test #2 should be useful, if no I think both tests aren't helpful anymore (since test #2 would always get undefined regardless of how hydration was used)

Reply:
So this test has been skipped since the move to Vite/`0.21`. @natemoo-re what should this test look for now?

Reply:
I believe `data-reactroot` is required for hydrated components still. If this test is broken it might be a symptom of a bug we need to fix. Happy to spin off into a separate PR.

Reply:
Sounds good. Yeah if that could be investigated separately (ideally testing dev and build) that would be good. This test needs to be updated, but I wasn’t sure exactly how (and is unrelated to this PR).